tests/mocha/block_test.js

```diff
@@ -2257,6 +2266,125 @@ suite('Blocks', function () {
           'block should be marked disabled because enabled json property was set to false',
         );
       });
+      test('Disabled blocks from XML should have proper disabled status', function () {
+        // Nested c-shaped blocks, inner block is disabled
+        const blockXml = `<xml xmlns="https://developers.google.com/blockly/xml">
+        <block type="controls_if" x="63" y="87">
+          <statement name="DO0">
+            <block type="controls_if" disabled="true"></block>
+          </statement>
+        </block>
+      </xml>`;
+        Blockly.Xml.domToWorkspace(
+          Blockly.utils.xml.textToDom(blockXml),
+          this.workspace,
+        );
+        // Inner block
+        const block = this.workspace.getTopBlocks(false)[0].getChildren()[0];
+        chai.assert.isTrue(
+          block.visuallyDisabled,
```

Comment:
visuallyDisabled is private. Don't know if that's going to cause us problems in the future.

Reply:
Hmm. This will cause problems if tests are ever in TS (we'll also have lots of other problems I would assume). I don't know if we'll ever do that - seems like stubbing, etc. would have tons of type problems.

Any suggestions? I don't really want to add a public API for this. You can't just check what classes it has for example, because the thing that was broken before was specifically that this property was incorrectly set despite being rendered correctly. We have the TEST_ONLY rigamarole but that was only when needed due to module shenanigans and it's not here. Have we decided anything about this before?

Reply:
Eh I think this is fine. I think we mostly just don't test private things. But it makes sense here cuz it's a complex behavior, but we want to keep it private b/c it's just for performance improvments.